packages/blocks/tests/php/test-blocks.php

```diff
@@ -50,10 +67,10 @@ public function test_block_classes() {
 	 */
 	public function test_is_amp_request() {
 		add_filter( 'jetpack_is_amp_request', '__return_true' );
-
-		$this->assertTrue( Blocks::is_amp_request() );
-
+		$is_amp = Blocks::is_amp_request();
 		remove_filter( 'jetpack_is_amp_request', '__return_true' );
+
+		$this->assertTrue( $is_amp );
```

Comment:
No objection here, just starting a discussion.

I can think of two other ways to do this. Is there any particular preference in Jetpack or WordPress code for this versus one of the others, or is it just up to whoever is writing the test?

One alternative uses try-finally:
``` lang=php
add_filter( 'jetpack_is_amp_request', '__return_true' );
try {
    $this->assertTrue( Blocks::is_amp_request() );
} finally {
    remove_filter( 'jetpack_is_amp_request', '__return_true' );
}
```
That works pretty well for something short like this, but if there's more code inside it can get to be a bit much. And even more so if there are nested sets of things that might need undoing.

Another option uses a neat little class called [ScopedCallback](https://packagist.org/packages/wikimedia/scoped-callback):
``` lang=php
add_filter( 'jetpack_is_amp_request', '__return_true' );
$reset = new ScopedCallback( 'remove_filter', array( 'jetpack_is_amp_request', '__return_true' ) );

$this->assertTrue( Blocks::is_amp_request() );
```
When `$reset` goes out of scope, the ScopedCallback object's destructor will call the callback to perform the reset. You can, of course, pass it a Closure if you need to do more than just call one function to do the cleanup. OTOH, the seemingly-unused variable might confuse people unfamiliar with it, and that particular package doesn't support obsolete versions of PHP anymore (although it looks like the code itself is fine).

Reply:
I don't think we really have an agreed-upon approach on those kinds of situations, it often depends on the person writing the tests. I like your try-finally proposal, it's clear and readable here. I've committed it in 17e8d4a1cf3ebc57e9a980e3389c8e23abd07b4b.